src/main/kotlin/com/bitclave/node/services/v1/OfferService.kt

```diff
@@ -115,6 +115,50 @@ class OfferService(
         })
     }
 
+    fun putBulkAdvanced(
+        owner: String,
+        offers: Array<Offer>,
+        strategy: RepositoryStrategyType
+    ): CompletableFuture<List<Long>> {
+        return supplyAsyncEx(Supplier {
+            val updatedOfferIds = offers.filter { it.id != 0L }.map { it.id }
+
+            val existedOffers = offerRepository.changeStrategy(strategy).findByIds(updatedOfferIds)
+
+            var index = 0
+            val readyForSaveOffers = offers.map {
+                val createdAt = if (it.id != 0L) existedOffers[index++].createdAt else Date()
```

Comment:
existedOffers may be have different size than offers. u should get item by offer id.

Reply:
fixed